### ISSUE TITLE
[Feature]: Refactoring the code

### ISSUE THREAD

**Issue by ppeigne** [AI-2-The-Sky/ResearchRL#7]
### Description

Simon's code is in a jupyter notebook. 
It must be put into .py files. 
The jupyter notebooks can be used for experimentation but the main code should not be put into them.

### Expected behaviour

1 file for each main component, containing (at least) one class. 
- [ ] Agent file: 
   - [ ] action selection
   - [ ] recording of state, action, reward, new_state, done (a dedicated Memory object could be a good idea)
   - [ ] training of the Q network 
      - [ ] collect of a batch of previous experiences
      - [ ] generation of the training signal
      - [ ] run a train of the Q net
    
- [ ] Q Network file
  - [ ] pytorch NN class

- [ ] Trainer file
  - [ ] training
  - [ ] testing
  - [ ] saving the params / hyper params 

- [ ] Preprocessing file (input data consists of arrays of string, it should be preprocessed before ingestion by the Q Net)
  - [ ] Convert the strings into numerical data
     - [ ] ordinal encoding ?
     - [ ] one hot encoding
  - [ ] Convert the encoded data into torch compatible data 

### Environment

- OS and version
